File: trunk/HeartRateMonitor.Server/HeartRateMonitor.Math/Evaluation/PNN50.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;

namespace HeartRateMonitor.Math.Evaluation
{
    public class PNN50 : IEvaluation<double>
    {
        public double Evaluate(SessionData session)
        {
            int pnn = 0;

            for (int i = 1, count = session.Intervals.Count; i < count; i++)
            {
                int now = session.Intervals[i];
                int before = session.Intervals[i - 1];
                if (System.Math.Abs(now - before) >= 50) {
                    pnn++;
                }
            }

            return ((double) pnn) / (session.Intervals.Count - 1);
        }
    }
}

File: trunk/PolarMath/src/org/macondo/polar/evaluation/PNN50.java
package org.macondo.polar.evaluation;

import org.macondo.polar.data.Training;

import java.util.List;

/**
 * <p></p>
 *
 * Date: 19.05.2008
 *
 * @author <a href="mailto:ktsibriy@gmail.com">Kirill Y. Tsibriy</a>
 */
public class PNN50 implements Evaluation<Double> {
    public Double evaluate(Training training) {
        final List<Integer> intervals = training.getIntervals();
        int pnn = 0;


        for (int i = 1; i < intervals.size(); i++) {
            Integer now = intervals.get(i);
            Integer before = intervals.get(i-1);

            if (Math.abs(now - before) >= 50) {
                pnn++;
            }
        }

        return ((double) pnn) / (intervals.size() - 1);
    }
}
